app/src/main/java/es/craftsmanship/toledo/katangapp/utils/AndroidBus.java

```diff
@@ -0,0 +1,39 @@
+package es.craftsmanship.toledo.katangapp.utils;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import com.squareup.otto.Bus;
+
+/**
+ * @author Javier Gamarra
+ */
+public class AndroidBus {
+
+    private static final Bus BUS = new MainThreadBus();
+
+    private AndroidBus() {
+    }
+
+    public static Bus getInstance() {
+        return BUS;
+    }
+
+    public static class MainThreadBus extends Bus {
+        private final Handler handler = new Handler(Looper.getMainLooper());
+
+        @Override
+        public void post(final Object event) {
+            if (Looper.myLooper() == Looper.getMainLooper()) {
```

Comment:
It's equivalent. We are checking to see if we are already executing code on the main thread so if the reference (unique) is the same, we are and we don't have to post a message with the handler.